tests/test_nested.py

```diff
@@ -25,6 +26,16 @@ def test_nested_list_of_dataclasses(self):
         assert (DataClassXs.from_json('{"xs": [{"x": 0}, {"x": 1}]}') ==
                 DataClassXs([DataClassX(0), DataClassX(1)]))
 
+    def test_nested_mapping_of_dataclasses(self):
+        err = None
+        try:
+            DataClassMappingBadDecode.from_dict(dict(map=dict(test=dict(id="irrelevant"))))
```

Comment:
I would recommend use `pytest.raises` for testing expected exceptions, here the docs: https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Reply:
i'm not terribly familiar with `pytest`, and i was hoping there would be something like that, thanks!